src/citrine/resources/job.py

```diff
@@ -1,132 +1,6 @@
-from logging import getLogger
-from typing import Union
-from uuid import UUID
+import citrine.jobs.job
+from citrine._utils.functions import shadow_classes_in_module
+import sys
 
-from time import time, sleep
 
-from citrine._serialization.properties import Set as PropertySet, String, Object
-from citrine._rest.resource import Resource
-from citrine._serialization import properties
-from citrine._session import Session
-from citrine.exceptions import PollingTimeoutError, JobFailureError
-
-logger = getLogger(__name__)
-
-
-class JobSubmissionResponse(Resource['AraJobStatus']):
-    """[ALPHA] a response to a submit-job request for the job submission framework.
-
-    This is returned as a successful response from the remote service.
-    """
-
-    job_id = properties.UUID("job_id")
-    """:UUID: job id of the job submission request"""
-
-    def __init__(self):
-        pass  # pragma: no cover
-
-
-class TaskNode(Resource['TaskNode']):
-    """[ALPHA] individual task status.
-
-    The TaskNode describes a component of an overall job.
-    """
-
-    id = properties.String("id")
-    """:str: unique identification number for the job task"""
-    task_type = properties.String("task_type")
-    """:str: the type of task running"""
-    status = properties.String("status")
-    """:str: The last reported status of this particular task.
-    One of "Submitted", "Pending", "Running", "Success", or "Failure"."""
-    dependencies = PropertySet(String(), "dependencies")
-    """:Set[str]: all the tasks that this task is dependent on"""
-    failure_reason = properties.Optional(String(), "failure_reason")
-    """:str: if a task has failed, the failure reason will be in this parameter"""
-
-    def __init__(self):
-        pass  # pragma: no cover
-
-
-class JobStatusResponse(Resource['JobStatusResponse']):
-    """[ALPHA] a response to a job status check.
-
-    The JobStatusResponse summarizes the status for the entire job.
-    """
-
-    job_type = properties.String("job_type")
-    """:str: the type of job for this status report"""
-    status = properties.String("status")
-    """:str: The status of the job. One of "Running", "Success", or "Failure"."""
-    tasks = properties.List(Object(TaskNode), "tasks")
-    """:List[TaskNode]: all of the constituent task required to complete this job"""
-    output = properties.Optional(properties.Mapping(String, String), 'output')
-    """:Optional[dict[str, str]]: job output properties and results"""
-
-    def __init__(self):
-        pass  # pragma: no cover
-
-
-def _poll_for_job_completion(session: Session, project_id: Union[UUID, str],
```

Comment:
Heads up that this method doesn't get shadowed, but it should be fine since it's a private method